src/HTTPure/Request.purs

```diff
@@ -51,12 +68,56 @@ fullPath request = "/" <> path <> questionMark <> queryParams
 -- | Given an HTTP `Request` object, this method will convert it to an HTTPure
 -- | `Request` object.
 fromHTTPRequest :: HTTP.Request -> Aff Request
-fromHTTPRequest request = pure
-  { method: Method.read request
-  , path: Path.read request
-  , query: Query.read request
-  , headers: Headers.read request
-  , body: Body.read request
-  , httpVersion: Version.read request
-  , url: requestURL request
-  }
+fromHTTPRequest request = do
+  body <-
+    liftEffect
+      $ Ref.new
+          { buffer: Nothing
+          , stream: Body.read request
+          , string: Nothing
+          }
+  pure
+    { method: Method.read request
+    , path: Path.read request
+    , query: Query.read request
+    , headers: Headers.read request
+    , body
+    , httpVersion: Version.read request
+    , url: requestURL request
+    }
+
+readBodyAsBuffer :: Request -> Aff Buffer
```

Comment:
These functions should all be changed to `Body -> Aff Buffer` and moved into `Body.purs` (in fact, you should probably just replace the current `toBuffer` / `toString` and add a `toStream` function as the current `toBuffer` and `toString` should probably be dropped from the public API entirely given this bug).

Reply:
Updated.